### ISSUE TITLE
[Content Block Card] Video - Screen reader focus is not trapped within the dialog

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#5670]
**Engineering info:**  
- Innovation Team engineer: @andy-blum 
- Innovation Team JIRA Ticket: https://jsw.ibm.com/browse/HC-2398
- DPO consulting engineer: N/A

-----

## Detailed description

JAWS FF/Voiceover iPhone

Issue:
1. In jaws FF, The screen reader focus is not trapped within the dialog. Screen reader focus is trapped in the modal in React staging
2. In Vo iPhone, the screen reader focus is not trapped in both the Environments

Expected:
Screen reader focus should be trapped within the modal dialog

**Comment by jkaeser** [comment]
@andysherman2121 Do you or the team have any recommendations for how to test this while using a machine running macOS?

**Comment by IgnacioBecerra** [comment]
@jkaeser We've done some JAWS bug fixes using a virtual machine, specifically VirtualBox. 

1. Download [Virtual Box](https://www.virtualbox.org/wiki/Downloads)
2. Go to https://developer.microsoft.com/en-us/microsoft-edge/tools/vms/ and choose the following
-Virtual Machine: MSEdge on Win10
-VM platform: VirtualBox
3. Move the downloaded VM outside the download folder and click on the `.ovf` file (should be orange)
4. The VM should open up in VirtualBox and you can click on start
5. You should see Windows 10 start up prompting for a password. Input: `Passw0rd!` 
6. Within the VM, go to the JAWS website and download it

Might have to tinker with some settings (can't verify since I had to delete mine some time ago), but this is how we can debug some JAWS issues. Hope it helps!

cc: @andysherman2121

**Comment by andy-blum** [comment]
Heads up a JAWS license is like $100/yr

**Comment by IgnacioBecerra** [comment]
Should've linked this in my comment, but this is the free 40 minute trial version of JAWS that we used: https://support.freedomscientific.com/Downloads/JAWS

You'll need to start JAWS first, and then open the browser. 

Once the 40 minutes are up you'll have to restart the VM to get it back up and running unfortunately.

**Comment by proeung** [comment]
@IgnacioBecerra Thanks for the info! As we're starting to see more of these accessibility issues coming from the QA team, I wonder if it makes sense for us to look into acquiring a "Business" tier license for JAWS so that we can debug the issue and verify the fix correctly. @jeffchew Let me know what you think?

Also, the steps outlined in this comment (https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5670#issuecomment-958405419) is quite helpful, is this something that we can include in the contribution docs (eg. see [link](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/master/docs/developing.md)) somewhere so that the devs working on this type of issue can look into using JAWS via VirtualBox?

**Comment by jkaeser** [comment]
Here are some notes that may be useful to whoever picks up this task:

- The code that is responsible for the modal is coming from the archived/deprecated **carbon-components-react** package, specifically https://github.com/carbon-design-system/carbon-components-react/blob/master/src/components/ComposedModal/ComposedModal.js
- There doesn't appear to be a substitute for `ComposedModal` in the replacement **carbon-react** package. Perhaps creating one is a blocker for this issue, or maybe we'll have to create a new substitute in this package.
